packages/editor-ui/src/mixins/externalHooks.ts

```diff
@@ -1,30 +1,167 @@
 import { IExternalHooks } from '@/Interface';
-import { useWebhooksStore } from '@/stores/webhooks';
-import { IDataObject } from 'n8n-workflow';
-import { Store } from 'pinia';
+import { IDataObject, INode, INodeTypeDescription } from 'n8n-workflow';
 import Vue from 'vue';
+import { Route } from 'vue-router/types/router';
+import {
+	AuthenticationModalEventData,
+	ExecutionFinishedEventData,
+	ExecutionStartedEventData,
+	ExpressionEditorEventsData,
+	InsertedItemFromExpEditorEventData,
+	NodeRemovedEventData,
+	NodeTypeChangedEventData,
+	OutputModeChangedEventData,
+	UpdatedWorkflowSettingsEventData,
+	UserSavedCredentialsEventData,
+} from '@/hooks/telemetry';
+
+export interface ExternalHooks {
+	parameterInput: {
+		mount: Array<
+			(meta: { inputFieldRef: { $el: HTMLElement }; parameter: { name: string } }) => void
+		>;
+	};
+	nodeCreatorSearchBar: {
+		mount: Array<(meta: { inputRef: HTMLElement }) => void>;
+	};
+	app: {
+		mount: Array<(meta: {}) => void>;
+	};
+	nodeView: {
+		mount: Array<(meta: {}) => void>;
+		createNodeActiveChanged: Array<(meta: { source: string }) => void>;
+		addNodeButton: Array<(meta: { nodeTypeName: string }) => void>;
+	};
+	main: {
+		routeChange: Array<(meta: { to: Route; from: Route }) => void>;
+	};
+	credential: {
+		saved: Array<(meta: UserSavedCredentialsEventData) => void>;
+	};
+	credentialsEdit: {
+		credentialTypeChanged: Array<
+			(meta: {
+				newValue: string;
+				setCredentialType: string;
+				credentialType: string;
+				editCredentials: string;
+			}) => void
+		>;
+		credentialModalOpened: Array<
+			(meta: { activeNode: INode; isEditingCredential: boolean; credentialType: string }) => void
+		>;
+	};
+	credentialsList: {
+		mounted: Array<(meta: {}) => void>;
+		dialogVisibleChanged: Array<(meta: { dialogVisible: boolean }) => void>;
+	};
+	workflowSettings: {
+		dialogVisibleChanged: Array<(meta: { dialogVisible: boolean }) => void>;
+		saveSettings: Array<(meta: UpdatedWorkflowSettingsEventData) => void>;
+	};
+	dataDisplay: {
+		onDocumentationUrlClick: Array<
+			(meta: { nodeType: INodeTypeDescription; documentationUrl: string }) => void
+		>;
+		nodeTypeChanged: Array<(meta: NodeTypeChangedEventData) => void>;
+		nodeEditingFinished: Array<(meta: {}) => void>;
+	};
+	executionsList: {
+		openDialog: Array<(meta: {}) => void>;
+	};
+	showMessage: {
+		showError: Array<(meta: { title: string; message: string; errorMessage: string }) => void>;
+	};
+	expressionEdit: {
+		itemSelected: Array<(meta: InsertedItemFromExpEditorEventData) => void>;
+		dialogVisibleChanged: Array<(meta: ExpressionEditorEventsData) => void>;
```

Comment:
`closeDialog` is missing
<img width="881" alt="Bildschirmfoto 2023-03-07 um 14 07 43" src="https://user-images.githubusercontent.com/4711238/223431526-7dd92052-e00f-4e69-84cd-19c68431a041.png">

Reply:
Added